### ISSUE TITLE
Error in registration when no config is provided

### ISSUE THREAD

**Issue by bailejl** [AnthonyNahas/ngx-auth-firebaseui#190]
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### OS and Version?

All

### Versions

ng --version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 7.2.4
Node: 11.0.0
OS: darwin x64
Angular: 7.2.5
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.13.1
@angular-devkit/build-angular     0.13.1
@angular-devkit/build-optimizer   0.13.1
@angular-devkit/build-webpack     0.13.1
@angular-devkit/core              7.3.1
@angular-devkit/schematics        7.2.4
@angular/cdk                      7.3.2
@angular/cli                      7.2.4
@angular/fire                     5.1.1
@angular/flex-layout              7.0.0-beta.23
@angular/material                 7.3.2
@ngtools/webpack                  7.3.1
@schematics/angular               7.2.4
@schematics/update                0.12.4
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.29.0
    

### Repro steps

1. Pass no config to NgxAuthFirebaseUIModule.forRoot();
2. Register with a username and password
3. Console shows error "Cannot read property 'enableFirestoreSync' of undefined"

### The log given by the failure

2019-02-17T03:52:32.854Z There was a Type error. Cannot read property 'enableFirestoreSync' of undefined
push../src/app/core/services/base/error-logging.service.ts.ErrorLoggingService.logError | @ | error-logging.service.ts:16
-- | -- | --
  | push../src/app/core/login/login.component.ts.LoginComponent.loginError | @ | login.component.ts:17
  | (anonymous) | @ | LoginComponent.html:2
  | handleEvent | @ | core.js:23097
  | callWithDebugContext | @ | core.js:24167
  | debugHandleEvent | @ | core.js:23894
  | dispatchEvent | @ | core.js:20546
  | (anonymous) | @ | core.js:22036
  | schedulerFn | @ | core.js:13517
  | push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub | @ | Subscriber.js:196
  | push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next | @ | Subscriber.js:134
  | push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next | @ | Subscriber.js:77
  | push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next | @ | Subscriber.js:54
  | push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next | @ | Subject.js:47
  | push../node_modules/ngx-auth-firebaseui/esm5/ngx-auth-firebaseui.es5.js.AuthProcessService$$1.handleError | @ | ngx-auth-firebaseui.es5.js:541
  | (anonymous) | @ | ngx-auth-firebaseui.es5.js:372
  | step | @ | ngx-auth-firebaseui.es5.js:156
  | (anonymous) | @ | ngx-auth-firebaseui.es5.js:137
  | fulfilled | @ | ngx-auth-firebaseui.es5.js:128

2019-02-17T03:52:32.868Z There was a general error. Uncaught (in promise): TypeError: Cannot read property 'toastMessageOnAuthError' of undefined
TypeError: Cannot read property 'toastMessageOnAuthError' of undefined
    at AuthProcessService$$1.push../node_modules/ngx-auth-firebaseui/esm5/ngx-auth-firebaseui.es5.js.AuthProcessService$$1.handleError (http://localhost:4200/vendor.js:188435:25)
    at AuthProcessService$$1.<anonymous> (http://localhost:4200/vendor.js:188265:30)
    at step (http://localhost:4200/vendor.js:188049:23)
    at Object.next (http://localhost:4200/vendor.js:188030:53)
    at fulfilled (http://localhost:4200/vendor.js:188021:58)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (http://localhost:4200/polyfills.js:2749:26)
    at Object.onInvoke (http://localhost:4200/vendor.js:78174:33)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (http://localhost:4200/polyfills.js:2748:52)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (http://localhost:4200/polyfills.js:2508:43)
    at http://localhost:4200/polyfills.js:3247:34

### Desired functionality

One of the following:
- Require a config object 
- Or have the code work without one.

### Mention any other details that might be useful

The code is some spots are existing configuration variables and not defaulting to a value.  Adding a config object got me around it.  However, by fixing this issue, others can save time.

**Comment by AnthonyNahas** [comment]
I will fix that asap and write unit tests for that to prevent those issues in the future! Thanks for reporting 🍺

**Comment by AnthonyNahas** [comment]
I've just patched the library and this is fixed in [v2.3.3](https://www.npmjs.com/package/ngx-auth-firebaseui)

thanks again 👍
